org.testeditor.web.backend.persistence/src/test/java/org/testeditor/web/backend/persistence/git/GitProviderTest.xtend

```diff
@@ -0,0 +1,73 @@
+package org.testeditor.web.backend.persistence.git
+
+import com.google.inject.Module
+import java.io.File
+import java.util.List
+import javax.inject.Inject
+import org.eclipse.jgit.api.Git
+import org.junit.Test
+import org.mockito.Mock
+import org.testeditor.web.backend.persistence.workspace.WorkspaceProvider
+
+import static org.eclipse.jgit.lib.ConfigConstants.*
+import static org.eclipse.jgit.lib.Constants.DEFAULT_REMOTE_NAME
+
+import static extension org.mockito.Mockito.*
+
+class GitProviderTest extends AbstractGitTest {
+
+	@Inject GitProvider gitProvider
+	@Mock WorkspaceProvider workspaceProvider
+
+	override protected collectModules(List<Module> modules) {
+		super.collectModules(modules)
+
+		// configure WorkspaceProvider mock
+		when(workspaceProvider.workspace).thenReturn(localGitRoot.root)
+		modules += [ binder |
+			binder.bind(WorkspaceProvider).toInstance(workspaceProvider)
+		]
+	}
+
+	@Test
+	def void clonesRemoteRepository() {
+		// when
+		val git = gitProvider.git
+
+		// then
+		getRemoteUrl(git).assertEquals('file://' + remoteGitFolder.root.absolutePath)
+		new File(localGitRoot.root, "README.md").exists.assertTrue
+		verify(workspaceProvider, 1.times).workspace
```

Comment:
`verify(workspaceProvider).workspace` should be sufficient, since `times(1)` is the default.